xbmc/pvr/PVRChannelGroupImageFileLoader.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PVRChannelGroupImageFileLoader.h"
+
+#include "FileItem.h"
+#include "filesystem/PVRGUIDirectory.h"
+#include "guilib/Texture.h"
+#include "pictures/Picture.h"
+#include "pvr/PVRCachedImages.h"
+#include "utils/log.h"
+
+bool PVR::CPVRChannelGroupImageFileLoader::CanLoad(const std::string& specialType) const
+{
+  return specialType == "pvr";
+}
+
+std::unique_ptr<CTexture> PVR::CPVRChannelGroupImageFileLoader::Load(const std::string& specialType,
+                                                                     const std::string& filePath,
+                                                                     unsigned int,
+                                                                     unsigned int) const
+{
+  const CPVRGUIDirectory channelGroupDir(filePath);
+  CFileItemList channels;
+  if (!channelGroupDir.GetChannelsDirectory(channels))
+  {
+    return {};
+  }
+
+  std::vector<std::string> channelIcons;
+  for (const auto& channel : channels)
+  {
+    const std::string& icon = channel->GetArt("icon");
+    if (!icon.empty())
+      channelIcons.emplace_back(CPVRCachedImages::UnwrapImageURL(icon));
+
+    if (channelIcons.size() == 9) // limit number of tiles
+      break;
+  }
+
+  return CPicture::CreateTiledThumb(channelIcons);
```

Comment:
The old implementation did much more after calling `CPicture::CreateTiledThumb`. Why is this no longer needed? Just asking.

```c++
    if (CPicture::CreateTiledThumb(channelIcons, CTextureCache::GetCachedPath(relativeCacheFile)))
    {
      CTextureDetails details;
      details.file = relativeCacheFile;
      details.width = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_imageRes;
      details.height = details.width;
      CServiceBroker::GetTextureCache()->AddCachedTexture(thumb, details);
      return thumb;
    }
```

Reply:
"Motivation: Consistent behavior" That code just manually saved one particular image type to the texture cache. Now all image types are saved to the texture cache in one place in the texture cache code after any image is loaded, no need to duplicate it.